File: playground/ruthetum/practice/oom/src/main/java/team/sipe/spurt/practice/oom/CountryService.java
package team.sipe.spurt.practice.oom;

import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CountryService {

    private CountryClient countryClient;

    public CountryService(CountryClient countryClient) {
        this.countryClient = countryClient;
    }

    public List<CountryResponse> getExternalCountry() {
        return countryClient.getExternalCountry()
                .stream()
                .map(CountryResponse::fromExternal)
                .toList();
    }
}

File: playground/ruthetum/practice/oom/src/main/kotlin/team/sipe/spurt/practice/oom/CountryService.kt
package team.sipe.spurt.practice.oom

import org.springframework.stereotype.Service

@Service
class CountryService {

    companion object {
        private const val API_URL = "https://gist.githubusercontent.com/anubhavshrimal/75f6183458db8c453306f93521e93d37/raw/f77e7598a8503f1f70528ae1cbf9f66755698a16/CountryCodes.json"
    }
}